services/permits/app/permit_conditions/resources/permit_condition_resource.py

```diff
@@ -42,56 +50,83 @@ async def extract_permit_conditions(file: UploadFile = File(...)) -> PermitCondi
 
     # Write the uploaded file to a temporary file
     # so it can be processed by the pipeline.
-    tmp = store_temporary(file)
+    tmp = store_temporary(file, suffix=".pdf")
 
     try:
-        pipeline = permit_condition_pipeline()
-
-        return pipeline.run(
-            {
-                "pdf_converter": {"file_path": tmp.name},
-                "prompt_builder": {
-                    "template_variables": {
-                        "max_pages": 6,
-                    }
-                },
-            }
-        )["validator"]
-    finally:
+        res = run_permit_condition_pipeline.delay(tmp.name, {"original_file_name": file.filename, 'size': file.size, 'content_type': file.content_type})
+        return JobStatus(id=res.task_id, status=res.status, meta=res.info)
+    except:
         tmp.close()
+        raise
+
+@router.get("/permit_conditions/status")
+def status(task_id: str) -> JobStatus:
+    """
+    Get the status of a permit conditions extraction job.
+    Args:
+        The task ID of the job.
+    """
+    res = run_permit_condition_pipeline.app.AsyncResult(task_id)
+    return JobStatus(id=res.task_id, status=res.status, meta=res.info)
+
+@router.get("/permit_conditions/results")
+def results(task_id: str) -> PermitConditions:
+    """
+    Get the results of a permit conditions extraction job.
+    Args:
+        The task ID of the job.
+    Raises:
+        400 Bad Request: If the task has not completed successfully
+    """
+    res = run_permit_condition_pipeline.app.AsyncResult(task_id)
 
+    if res.status == "SUCCESS":
+        return res.get()
+    else:
+        raise HTTPException(400, detail=f"Task has not completed successfully. Current status: {res.status}")
```

Comment:
I think I'd expect to get a 400 if I sent a bad task id, but I wonder if there's something more appropriate to send back to indicate "still working on it". Maybe `202 Accepted`? Or a 500 if it's a bad status?

Reply:
Good call, 202 seems good for the "processing" stage of it.
I updated it to be:
`200` -> Task finished
`202` -> Task created, but not finished
`500` -> Task failed